netty/src/test/java/io/grpc/netty/NettyServerStreamTest.java

```diff
@@ -280,14 +281,39 @@ public void cancelStreamShouldSucceed() {
         true);
   }
 
+  @Test
+  public void oversizedMessagesResultInResourceExhaustedTrailers() throws Exception {
+    @SuppressWarnings("InlineMeInliner") // Requires Java 11
+    String oversizedMsg = Strings.repeat("a", TEST_MAX_MESSAGE_SIZE + 1);
+    stream.request(1);
+    stream.transportState().inboundDataReceived(messageFrame(oversizedMsg), false);
+    assertNull("message should have caused a deframer error", listenerMessageQueue().poll());
+
+    ArgumentCaptor<SendResponseHeadersCommand> sendHeadersCap =
+            ArgumentCaptor.forClass(SendResponseHeadersCommand.class);
+    verify(writeQueue).enqueue(sendHeadersCap.capture(), eq(true));
+
+    Status status = Status.RESOURCE_EXHAUSTED
+            .withDescription("gRPC message exceeds maximum size 128: 129");
+
+    SendResponseHeadersCommand actualCmd = sendHeadersCap.getValue();
+    assertThat(actualCmd.status().getCode()).isEqualTo(status.getCode());
+    assertThat(actualCmd.status().getDescription()).isEqualTo(status.getDescription());
+    Metadata trailers = Utils.convertTrailers(actualCmd.headers());
+    assertThat(trailers.get(InternalStatus.CODE_KEY).getCode()).isEqualTo(status.getCode());
+    assertThat(trailers.get(InternalStatus.MESSAGE_KEY)).isEqualTo(status.getDescription());
+  }
+
+  private static final int TEST_MAX_MESSAGE_SIZE = 128;
```

Comment:
minor: the static global parameter shouble be on the top of class ?

Reply:
SG, done